bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineManagerImpl.java

```diff
@@ -216,12 +222,14 @@ public void removeEngine(String engineIdentifier) {
             }
 
             if (scriptEngine instanceof AutoCloseable) {
-                AutoCloseable closeable = (AutoCloseable) scriptEngine;
-                try {
-                    closeable.close();
-                } catch (Exception e) {
-                    logger.error("Error while closing script engine", e);
-                }
+                scheduler.schedule(() -> {
```

Comment:
I didn't use `.execute` here because the JavaDoc says, `.execute` may use the calling thread to execute the task. That is exactly what we want to avoid here.

Reply:
Sounds like a good comment in the code. :wink: